### PR TITLE
Allow for early termination of the visitor

### PR DESCRIPTION
**Pull request by LukasPietzschmann**

### CODE REVIEW

File: R/visit.R [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #' Visits the set of nodes and all of their children, invoking the callback for each visited node
 #'
 #' @param nodes The list or array of nodes to visit
-#' @param callback The callback function to invoke for each node
+#' @param callback The callback function to invoke for each node. The callback should return TRUE to continue visiting the children of the node, or FALSE to stop visiting the children of the node.
```
Comment (EagleoutIce) [inline R/visit.R:4]: Maybe extend this to "anything non-false"? Or do we want a check to enforce `\(n) { magic(n); return TRUE }`

**Review by EagleoutIce — APPROVED** [review]

File: R/visit.R (hunk 1 above)
Comment (LukasPietzschmann) [reply R/visit.R]: Nah, I don't like requiring an explicit return to indicate default behavior. I fixed the doc comment.

**Comment by EagleoutIce** [comment]
I am fine with this! You as well?

**Comment by LukasPietzschmann** [comment]
> You as well?

Who is _you_? Me?

**Comment by EagleoutIce** [comment]
Yes :laughing:

**Comment by LukasPietzschmann** [comment]
Ah lol. Yeah, with bd28342 I'm fine. Aren't we waiting for @Ellpeck's approval?

**Comment by EagleoutIce** [comment]
Yes :)

**Review by Ellpeck — APPROVED** [review]

**Pull request merged**